File: 0070-ClimbingStairs/Solution.java
import java.util.*;

public class Solution {
    
    static int climbStairs(int n) {
        if (n < 3) {
            return n;
        }
        int prev = 1;
        int curr = 2;
        for (int i = 3; i <= n; i++) {
            int tmp = curr;
            curr = prev + curr;
            prev = tmp;
        }
        return curr;
    }

    public static void main(String[] args) {
        System.out.println();
    }

}

File: 0070-ClimbingStairs/solution.py
class Solution:

    def climb_stairs(self, n: int) -> int:
        if n < 3:
            return n
        prev = 1
        curr = 2
        for i in range(3, n + 1): # inclusive
            prev, curr = curr, prev + curr
        return curr

if __name__ == "__main__":
    s = Solution()

